server/routes/tickets.js

```
@@ -39,10 +40,63 @@ app.get('/:id', async(req, res) => {
     let id = fromParam(req, 'id');
     try {
         let result = await controller.getTicket(id);
+        if (result == null) {
+            res.status('404').json({
+                message: messages.NOT_FOUND
+            });
+        }
         res.send(result);
     } catch (err) {
-        res.send(err)
+        //maybe create error log file instead of send error
+        let message = messages.errors(err.errno);
+        res.status('500').json({
+            message: message,
+        });
+
+    }
+});
+
+app.post('/', async(req, res) => {
+    let body = fromBody(req);
+    try {
+        let result = await controller.createTicket(body);
+
+        if (result == null) {
+            res.status('404').json({
+                message: messages.NOT_FOUND
+            });
+
+        }
+
+        res.status(201).send(result);
+
+    } catch (err) {
+        //maybe create error log file instead of send error
+        let message = messages.errors(err.errno);
+        res.status('500').json({
+            message: message,
+        });
     }
 });
 
+app.patch('/:id', async(req, res) => {
+    let body = fromBody(req);
+    try {
+        let result = await controller.updateTicket(body, req.params.id);
+        if (result == null) {
```

I'm not sure this is correct. What happens if we get `null` for a reason different that the resource doesn't exist? Actually, we should not return null in that case. That's what exceptions are for.

---

could you please  give me an example?

---

The controller should throw an exception when something is not right. A create or update should not return null to signal an error,